pkgs/by-name/st/streamripper/package.nix

```
@@ -18,6 +19,21 @@ stdenv.mkDerivation rec {
     sha256 = "0hnyv3206r0rfprn3k7k6a0j959kagsfyrmyjm3gsf3vkhp5zmy1";
   };
 
+  patches = [
+    (fetchDebianPatch {
+      inherit pname version;
+      debianRevision = "2";
+      patch = "1075541-gcc14";
+      hash = "sha256-30bz7CDmbq+Bd+jTKSq7aJsXUJQAQp3nnJZvt3Qbp8Q=";
+    })
+    (fetchDebianPatch {
+      inherit pname version;
+      debianRevision = "2";
+      patch = "873964-http";
+      hash = "sha256-D6koUCbnJHtRuq2zZy9VrxymuGXN1COacbQhphgB8qo=";
+    })
```

```suggestion
    # fix build with gcc 14
    (fetchDebianPatch {
      inherit pname version;
      debianRevision = "2";
      patch = "1075541-gcc14";
      hash = "sha256-30bz7CDmbq+Bd+jTKSq7aJsXUJQAQp3nnJZvt3Qbp8Q=";
    })
    # <functionality>
    (fetchDebianPatch {
      inherit pname version;
      debianRevision = "2";
      patch = "873964-http";
      hash = "sha256-D6koUCbnJHtRuq2zZy9VrxymuGXN1COacbQhphgB8qo=";
    })
```

Always make it obvious why the patch is needed, either by comment or by name